dev/NumberBox/NumberBox.cpp

```diff
@@ -0,0 +1,387 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "common.h"
+#include "NumberBox.h"
+#include "NumberBoxAutomationPeer.h"
+#include "RuntimeProfiler.h"
+#include "ResourceAccessor.h"
+#include "Utils.h"
+
+NumberBox::NumberBox()
+{
+    __RP_Marker_ClassById(RuntimeProfiler::ProfId_NumberBox);
+
+    // Default values for the number formatter
+    auto formatter = winrt::DecimalFormatter();
+    formatter.IntegerDigits(1);
+    formatter.FractionDigits(0);
+    NumberFormatter(formatter);
+
+    PointerWheelChanged({ this, &NumberBox::OnScroll });
+
+    SetDefaultStyleKey(this);
+}
+
+winrt::AutomationPeer NumberBox::OnCreateAutomationPeer()
+{
+    return winrt::make<NumberBoxAutomationPeer>(*this);
+}
+
+void NumberBox::OnApplyTemplate()
+{
+    winrt::IControlProtected controlProtected = *this;
+
+    if (auto spinDown = GetTemplateChildT<winrt::RepeatButton>(L"DownSpinButton", controlProtected))
```

Comment:
>DownSpinButton [](start = 65, length = 14)

I'd prefer the names be pull out into a static constexpr

Reply:
fixed

Reply:
I still see a lot of locals which aren't const

---
In reply to: [341247515](https://github.com/microsoft/microsoft-ui-xaml/pull/1507#discussion_r341247515) [](ancestors = 341247515)

Reply:
I sprinkled the word "const" throughout the code.

Reply:
Thanks! I feel much more constant

---
In reply to: [341353871](https://github.com/microsoft/microsoft-ui-xaml/pull/1507#discussion_r341353871) [](ancestors = 341353871)